apps/aecore/lib/aecore/channel/tx/channel_deposit_tx.ex

```diff
@@ -221,8 +221,14 @@ defmodule Aecore.Channel.Tx.ChannelDepositTx do
       !ChannelStateOnChain.active?(channel) ->
         {:error, "#{__MODULE__}: Can't deposit from inactive channel."}
 
+      !ChannelStateOnChain.is_peer?(channel, depositing_account) ->
```

Comment:
Why do you move this check from `ChannelStateOnChain.validate_deposit` to transaction?

Reply:
In order to avoid passing the sender to the channel validation logic as it isn't used extensively in the on chain channel logic.